pkg/controller/watermarkpodautoscaler/replica_calculator.go

```diff
@@ -190,36 +204,67 @@ func getReplicaCount(logger logr.Logger, currentReplicas int32, wpa *v1alpha1.Wa
 	return replicaCount, utilizationQuantity.MilliValue()
 }
 
-func groupPods(logger logr.Logger, podList *v1.PodList, metrics metricsclient.PodMetricsInfo, resource v1.ResourceName, delayOfInitialReadinessStatus time.Duration) (readyPods, ignoredPods sets.String) {
+func (c *ReplicaCalculator) getReadyPodsCount(namespace string, selector labels.Selector, readinessDelay time.Duration) (int32, error) {
+	podList, err := c.podLister.Pods(namespace).List(selector)
+	if err != nil {
+		return 0, fmt.Errorf("unable to get pods while calculating replica count: %v", err)
+	}
+
+	if len(podList) == 0 {
+		return 0, fmt.Errorf("no pods returned by selector while calculating replica count")
+	}
+
+	toleratedAsReadyPodCount := 0
+
+	for _, pod := range podList {
+		_, condition := getPodCondition(&pod.Status, corev1.PodReady)
```

Comment:
Should we check instead: `PodAvailable` instead? (at least ready once)

Reply:
finally I think your code is looking as I say for pod ready at least once. since you donc check the `condition.Status == true`

Reply:
PodAvailable is not available, I think PodReady here is already a good indicator as it means that the pod is able to service requests and should be added to the load balancing pools of all matching services.